WallsAndHoles/tilemap.h

```diff
@@ -46,13 +47,32 @@ class TileMap : public QObject
     //changes the size of the map. If the size is reduced, tiles will be lost (resizes around top left corner)
     void resizeMap(QSize newSize);
 
+    const QString savePath() const { return mSavePath; }
+    void setSavePath(QString path){ mSavePath = path; }
+
+    //If tile-template link present, update tile id accordingly, otherwise create link according to tile id.
+    void setDepend(SharedTileTemplateSet templateSet){ mDependencies.push_back(templateSet); }
+    //remove all dependencies that are no longer used in this tilemap.
+    void updateDepend();
+
+    QVector<SharedTileTemplateSet> dependencies() const { return mDependencies; }
+
+    Array2D<QSharedPointer<Tile>> &tiles() { return mMap; }
```

Comment:
You don't need either of these. Just use `tileAt(...)` and `cTileAt(...)` and `mapSize()`